modules/microshift-adding-containers-to-blueprint.adoc

```diff
@@ -0,0 +1,59 @@
+// Module included in the following assemblies:
+//
+// microshift/microshift-embed-into-rpm-ostree.adoc
+
+[id="adding-{product-title}-container-images_{context}"]
+= Adding the {product-title} container images
+
+You can embed {product-title}'s container images into the {op-system-ostree} images so that they are immediately available to the CRI-O container engine after booting. By embedding, images are not pulled over the network from a container registry. In Image Builder, a container image is embedded by adding a reference to it to the Image Builder blueprint.
+
+.Example adding a container image to a blueprint:
+[source,terminal]
+----
+[[containers]]
+source = "<image_pullspec_with_tag_or_digest>"
+----
+
+You need to have the exact list of container image references used by the {product-title} version to embed {product-title} 's container images. Use the `microshift-release-info` RPM package matching the version of the `microshift` RPM in your blueprint.
+
+.Procedure
+
+. Download the `microshift-release-info` RPM package matching your {product-title} version by running the following commands:
++
+[source,terminal]
+----
+$ VERSION=$(rpm -q --queryformat='%{VERSION}-%{RELEASE}.%{ARCH}' microshift)
+----
++
+[source,terminal]
+----
+$ yum download microshift-release-info-${VERSION}
+----
+
+. Extract the release info by running the following command:
++
+[source,terminal]
+----
+$ rpm2cpio microshift-release-info-${VERSION}.rpm | cpio -idmv
+./usr/share/microshift/release/release-aarch64.json
+./usr/share/microshift/release/release-x86_64.json
+----
+
+. Generate the lines to append to your blueprint using the release info for your CPU architecture by running the following command:
++
+[source,terminal]
+----
+$ jq -r '.images | .[] | ("[[containers]]\nsource = \"" + . + "\"\n")' ./usr/share/microshift/release/release-$(uname -i).json
```

Comment:
According to [the repo guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output), we are not supposed to use `jq` unless there is no other alternative. If cases where `jq` must be used, it needs to be included in the prereqs.

Reply:
This command cannot be performed with `oc` by itself. It is using `jq` to read a JSON file and reformat it in the TOML structure needed for the image builder blueprint file. If documenting steps using jq is a problem we could consider an RFE to deliver the information in the TOML format in a future release, I guess, but this is what we have that works today.

Reply:
Thank you for clarifying. I understand that this is developer preview. In that case, we just need to add `jq` to the prerequisites. There are procedures that use `jq`, PX has just asked us to minimize the use when there is another way. Thanks for clarifying!

Reply:
@michaelryanpeter How do I incorporate this into the prereqs? Do I simply need to say something like, "you must be about to `jq`?